applications/GeoMechanicsApplication/custom_conditions/T_normal_flux_condition.cpp

```diff
@@ -0,0 +1,132 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+// Application includes
+#include "custom_conditions/T_normal_flux_condition.h"
+#include "utilities/math_utils.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::TNormalFluxCondition()
+    : TCondition<TDim, TNumNodes>()
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::TNormalFluxCondition(IndexType NewId,
+                                                            GeometryType::Pointer pGeometry)
+    : TCondition<TDim, TNumNodes>(NewId, pGeometry)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::TNormalFluxCondition(IndexType NewId,
+                                                            GeometryType::Pointer pGeometry,
+                                                            PropertiesType::Pointer pProperties)
+    : TCondition<TDim, TNumNodes>(NewId, pGeometry, pProperties)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::~TNormalFluxCondition() = default;
+
+template <unsigned int TDim, unsigned int TNumNodes>
+Condition::Pointer TNormalFluxCondition<TDim, TNumNodes>::Create(
+    IndexType NewId, NodesArrayType const& rThisNodes, PropertiesType::Pointer pProperties) const
+{
+    return Condition::Pointer(new TNormalFluxCondition(
+        NewId, this->GetGeometry().Create(rThisNodes), pProperties));
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TNormalFluxCondition<TDim, TNumNodes>::CalculateRHS(VectorType& rRightHandSideVector,
+                                                         const ProcessInfo& rCurrentProcessInfo)
+{
+    const GeometryType& rGeom = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& IntegrationPoints =
+        rGeom.IntegrationPoints(this->GetIntegrationMethod());
+    const unsigned int NumGPoints = IntegrationPoints.size();
+    const unsigned int LocalDim = rGeom.LocalSpaceDimension();
+
+    const Matrix& NContainer = rGeom.ShapeFunctionsValues(this->GetIntegrationMethod());
+    GeometryType::JacobiansType JContainer(NumGPoints);
+
+    for (auto& x : JContainer) {
+        x.resize(TDim, LocalDim, false);
+    }
+
+    rGeom.Jacobian(JContainer, this->GetIntegrationMethod());
+
+    array_1d<double, TNumNodes> normalFluxVector;
```

Comment:
Minor style guide thing:
```suggestion
    array_1d<double, TNumNodes> normal_flux_vector;
```

Reply:
changed